src/components/MessageBubble.js

```diff
@@ -1,17 +1,78 @@
-import React from 'react';
-import { auth } from '../firebase';
+import { auth, db } from '../firebase';
+import { useState } from 'react';
 import { useAuthState } from 'react-firebase-hooks/auth';
+import MessageEdit from './MessageEdit';
+import { updateDoc, doc } from 'firebase/firestore';
+
 const MessageBubble = ({ msg }) => {
+  // Getting the authenticated user using the 'useAuthState' hook
   const [user] = useAuthState(auth);
-  console.log(msg.user);
+  // Setting up a state to manage whether the edit message is visible or not
+  const [showEdit, setShowEdit] = useState(false);
+
+  // Event handler for clicking the edit button
+  const handleEditClick = () => {
+    setShowEdit(!showEdit);
+  };
+
+  // Function to update the messages content in the database
+  const messageUpdate = async (id, text) => {
+    const data = {
+      text: text,
+    };
+    const messageRef = doc(db, 'messages', id);
+    updateDoc(messageRef, data);
+
+    setShowEdit(!showEdit);
+  };
+
+  // Conditionally rendering the MessageEdit component if 'showEdit' is true
+  let content;
+  if (showEdit) {
+    content = <MessageEdit msg={msg} onSubmit={messageUpdate} />;
+  }
+
+  // Function to convert the message creation time to a formatted time string since firestore gives me time in seconds and milliseconds
+  function timeConvert(time) {
+    const milliseconds = time?.seconds * 1000 + time?.nanoseconds / 1000000;
+    const createdAtDate = new Date(milliseconds);
+    const options = {
+      hour12: true,
+      hour: 'numeric',
+      minute: 'numeric',
+      milliseconds: false,
+    };
```

Comment:
Amazing!  You were able set up a time stamp on the messages.  I can see it update from "just now" to the time of when the message was sent when a new message comes in.